### DISCUSSION TITLE
Alpine persist dynamic custom key

### DISCUSSION THREAD

**Discussion by keizah7** [alpinejs/alpine#3430] [7. Bugs (archived)]
```
export default (tabs = [], name = 'x', defaultTab = 0) => ({
    index: Alpine.$persist(defaultTab).as('tab-' . name),
});
```

```
x-data="tab([], 'x')">
```
```
x-data="tab([], 'b')">
```

For some reason local storage always have only `_x_index`, cant make it work with dynamic name

**Comment by keizah7** [comment]
It works without string concatenation.
```
    index: Alpine.$persist(defaultTab).as(name),
```

**Comment by ekwoka** [comment]
Your example is invalid js. Well, it is VALID, but it does not do what you want.

string concatenation in JS would be `'tab-' + name` or as a template literal ``` `tab-${name}` ```

`'tab-' . name` is trying to access the property `name` on the `String` `'tab-'` which, of course, doesn't exist.

**Reply by keizah7** [reply]
Oh, my bad :D
